src/box/lua/config/applier/box_cfg.lua

```diff
@@ -268,7 +268,8 @@ local function apply(config)
             if v ~= box.cfg[k] then
                 local warning = 'box_cfg.apply: non-dynamic option '..k..
                     ' will not be set until the instance is restarted'
-                config:_alert({type = 'warn', message = warning})
+                config:_alert('box_cfg_apply_non_dynamic',
+                              {type = 'warn', message = warning})
```

Comment:
Nit: I would add a registry of these keys and access it like so:

```lua
local alert_class = config.alert_class.BOX_CFG_APPLY_NON_DYNAMIC
config:_alert(alert_class, <...>)
```

There are only three classes now, so it is up to you.

Alternative: pass the alert class by name, but check it in `config:_alert()` against the registry.

Feel free to close this comment without actions.